Sources/RTVIClientIOSGeminiLiveWebSocket/util/AudioPlayer.swift

```diff
@@ -54,32 +48,40 @@ class AudioPlayer {
         try audioEngine.start()
         try playerNode.play()
         
-        didSetup = true
+        isRunning = true
     }
     
     func stop() {
-        if !didSetup { return }
+        if !isRunning { return }
         AudioCommon.uninstallAudioLevelTap(onNode: playerNode)
         playerNode.stop()
         enqueuedBufferCount = 0
         audioEngine.stop()
-        didSetup = false
+        isRunning = false
     }
     
     // TODO: maybe someday be smarter so changing devices doesn't cut off current output
     func adaptToDeviceChange() throws {
+        if !isRunning { return } // TODO: weird, why is audioEngine.isRunning false??
```

Comment:
Is it intentional, should we keep the comment ? 😅

Reply:
Whoops. Will remove. Good catch!